### PR TITLE
Rui/connect external program

### PR DESCRIPTION
**Pull request by bbaayezi**
### Purpose

Adds tutorial on how to integrate quantum flow with external program

### Changes

Slight changes in `quantum-circuit` node, `quantum-register` node, automatically check if the input message contains `binarString` field, if so, initilise the quantum circuit according to the input binary string.

### CODE REVIEW

**Comment by bbaayezi** [comment]
grover's test failed probably because of the lack of computing resource on CI servers. Local test suit will pass:
image
image

**Comment by louislefevre** [comment]
> grover's test failed probably because of the lack of computing resource on CI servers. Local test suit will pass:
> image
> image

I reran the workflow and it managed to finish in time on the second go. I fixed this in PR #88 by increasing the timeout period, so should no longer be an issue once that's merged 👍

**Comment by ttoktassynov** [comment]
Will you changes will fix this issue? #108

File: nodes/quantum/quantum-register/quantum-register.js [hunk 1]
```diff
@@ -122,6 +128,16 @@ module.exports = function(RED) {
       // wait for quantum circuit to be initialised
       logger.trace(node.id, 'Quantum register waiting for circuit to be ready');
       await circuitReady();
```
Comment (Theo-Reignier) [inline nodes/quantum/quantum-register/quantum-register.js:130]: @bbaayezi I just realized that one block of code in the register nodes might be useless.

When I first wrote the register node js files, I had to convey 2 validations:

1. Ensure all registers are initialized before running `qc = QuantumCircuit` 
2. Ensure all register nodes have finished running before quantum registers output qubit objects

Now, here is an animation that illustrates the order of execution of the nodes in Node-RED:
https://nodered.org/blog/content/images/2019/08/flow2-async.svg 

Following the order of execution, I think these 2 validations can be simplified a lot:

1. The last output of 'QuantumCircuit' will be the last register to be initialized. Therefore, we can just pass a variable to the last output of 'QuantumCircuit' that tells the 'register' node to initialise `qc`  .
2. The second validation is not even required anymore, the qubits objects will always be evaluated after the register nodes & `qc` are all initialised.

Let me know if my message wasn't clear !!

**Review by ttoktassynov — APPROVED** [review]
LGTM

File: nodes/quantum/quantum-register/quantum-register.js (hunk 1 above)
Comment (bbaayezi) [reply nodes/quantum/quantum-register/quantum-register.js:130]: Thanks, will look into it and create another PR for this.

**Pull request merged**